scripts/check_frontend_test_coverage.py

```diff
@@ -0,0 +1,407 @@
+    """
+    if not test_path:
+        sys.stderr.write(
+            'The test path is empty or null.'
+            'It\'s not possible to diff the test coverage correctly.')
+        sys.exit(1)
+
+    _, file_name = os.path.split(test_path)
+    return file_name
+
+
+def get_stanzas_from_lcov_file(file_path):
+    """Get all stanzas from a lcov file. Each stanza has the following
+    structure:
+    TN: test name
+    SF: file path
+    FNF: total functions
+    FNH: functions covered
+    LF: total lines
+    LH: lines covered
+    BRF: total branches
+    BRH: branches covered
+    end_of_record
+
+    This function filters each stanza to return only:
+    - File path (SF)
+    - File total lines (LF)
+    - File covered lines (LH)
+
+    Args:
+        file_path: str. The path of lcov file.
+
+    Returns:
+        list(str). A list with all stanzas filtered.
+    """
+    f = python_utils.open_file(file_path, 'r')
+    stanzas_array = f.read().split('end_of_record')
+    stanzas_array_filtered = []
+
+    for stanza in stanzas_array:
+        lines = [line for line in stanza.splitlines() if filter_lines(line)]
+        if len(lines) > 0:
+            stanzas_array_filtered.append(lines)
+
+    return stanzas_array_filtered
+
+
+def get_coverage_dict():
+    """Build a dict with all covered files from the current branch.
+
+    Returns:
+        Dictionary. A dict containing file path, total lines and covered lines
+        of each tested file.
+
+    Raises:
+        Exception: If DEV_LCOV_FILE_PATH doesn't exist.
+    """
+    coverage_dict = {}
+
+    if not os.path.exists(PR_LCOV_FILE_PATH):
+        raise Exception(
+            'File at path {} doesn\'t exist'.format(PR_LCOV_FILE_PATH))
+
+    stanzas = get_stanzas_from_lcov_file(PR_LCOV_FILE_PATH)
+    for stanza in stanzas:
+        total_lines = re.match('(\d+)', stanza[1].split(':')[1]).group(1)
+        covered_lines = re.match('(\d+)', stanza[2].split(':')[1]).group(1)
+
+        test_name = get_test_file_name(stanza[0])
+        coverage_dict[test_name] = [
+            int(total_lines),
+            int(covered_lines)
+        ]
+
+    return coverage_dict        
+
+
+def remove_item_from_whitelist(test_name):
+    """Remove an item from whitelist list.
+    
+    Args:
+      test_name: str. The test to be removed from whitelist.
+    """
+    index = fully_covered_tests.index(test_name)
+    fully_covered_tests.pop(index)
+
+
+def check_coverage_changes():
+    """Checks if the whitelist for fully covered files needs to be changed by:
+    - New file insertion
+    - File renaming
+    - File deletion
+
+    Raises:
+        Exception: If PR_LCOV_FILE_PATH doesn't exist.
+    """
+    if not os.path.exists(PR_LCOV_FILE_PATH):
+        raise Exception('File at path {} doesn\'t exist'.format(
+            PR_LCOV_FILE_PATH))
+
+    covered_tests = get_coverage_dict()
+    errors = ''
+
+    for test_name in covered_tests:
+        total_lines = covered_tests[test_name][0]
```

Comment:
You can do:

`total_lines, covered_lines = covered_tests[test_name]`

Reply:
I've refactored this piece of code, this doesn't exist anymore.